client/filestore-client/src/CordovaFileUpload.ts

```diff
@@ -0,0 +1,62 @@
+import * as Bluebird from 'bluebird';
+
+// Cordova mocks for typescript
+declare var FileUploadOptions: any;
+declare var FileTransfer: any;
+
+// Handling file upload to server
+function fileUpload(fileURI, serverURI, fileUploadOptions) {
+  const transfer = new FileTransfer();
+  return new Bluebird(function(resolve, reject) {
+    transfer.upload(fileURI, serverURI, function(response) {
```

Comment:
@JameelB This will fail currently as file transfer needs to work with passport/keycloak. 


We can pass headers here but this will mean that there will be 2 different places to edit those.
See: http://ngcordova.com/docs/plugins/fileTransfer/

@JameelB - What we should do here in your opinion:

Should we move this to the top level app to have control over the the options and allow users to control this?

Reply:
Some discussion around this topic in https://github.com/feedhenry-raincatcher/raincatcher-core/pull/159#discussion_r146268198.

tl;dr, we could ditch the cordova file transfer plugin and just have this module receive a function that'll handle the file upload itself.

Reply:
@paolobueno That will be the best aproach as we can put that on http module or even use file transfer directly in the application!

Reply:
@wtrocki @paolobueno if its possible to use angular-http i vote for doing it that way